pygmt/tests/test_figure.py

```diff
@@ -142,8 +142,7 @@ def test_figure_show():
     """
     fig = Figure()
     fig.basemap(region="10/70/-300/800", projection="X3i/5i", frame="af")
-    img = fig.show(width=800)
-    assert img.width == 800
+    fig.show()
```

Comment:
I can't revert the change. The old `Figure.show()` returns the Image object, but the new `Figure.show()` doesn't.

Reply:
Ah I see. Would it be possible to have the Image object be returned for `method='notebook'` and `None` for `method='external'`? Otherwise this will be a bit of a backward incompatible break.

Reply:
Yes, it's possible, but why do we want `Figure.show()` to return an Image object? What are the use cases?

Reply:
For plotting the Image in another Jupyter notebook cell? Or plotting it again and again without having to call `fig.show()` (which calls `psconvert`). The [IPython.display.Image](https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html) object also has other methods (e.g. `data` which gives raw image data) that could be used.